File: java/v1/src/main/java/com/api/v1/services/rent/AllRentsRetrievalService.java
package com.api.v1.services.rent;

import com.api.v1.dtos.rent.RentResponseDto;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface AllRentsRetrievalService {

    Mono<RentResponseDto> findByOrderNumber(String orderNumber);
    Flux<RentResponseDto> findAll();

}

File: kotlin/v2/src/main/kotlin/com/api/v2/rent/service/AllRentsRetrievalService.kt
package com.api.v2.rent.service

import com.api.v2.rent.dtos.RentResponseDto
import kotlinx.coroutines.flow.Flow

interface AllRentsRetrievalService {

    suspend fun findByOrderNumber(orderNumber: String): RentResponseDto
    suspend fun findAll(): Flow<RentResponseDto>

}